src/backend/commands/indexcmds.c

```diff
@@ -2819,7 +2819,8 @@ ReindexTable(ReindexStmt *stmt)
 	 */
 	if (get_rel_relkind(heapOid) == RELKIND_PARTITIONED_TABLE)
 	{
-		// FIXME transasction block check?
+		if (Gp_role == GP_ROLE_DISPATCH)
```

Comment:
I wonder why do we use `PreventInTransactionBlock()` only for `GP_ROLE_DISPATCH`  role here and in REINDEX_OBJECT_DATABASE? Don't we have the same problem in utility mode as well? In this case GPDB instance works in non-cluster mode and its behaviour should be equivalent to PostgreSQL, where we always call `PreventInTransactionBlock()`.